### ISSUE TITLE
Is there an incompatibility between unattended installs and setting Umbraco.Core.MainDom.Lock?

### ISSUE THREAD

**Issue by antonysmith-mando** [umbraco/Umbraco-CMS#11734]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

 (upgrading to) 8.15.3

### Bug summary

When trying to run an unattended upgrade with Umbraco.Core.MainDom.Lock set to SqlMainDomLock, I receive errors that Umbraco cannot start. The logs show errors regarding not being able to acquire the MainDom:

Umbraco.Core.Exceptions.BootFailedException: Boot failed: Umbraco cannot run. See Umbraco's log file for more details.\n\n-> Umbraco.Core.Exceptions.BootFailedException: Boot failed.\n\n-> System.TimeoutException: Cannot acquire MainDom

### Specifics

I had an instance of Umbraco running on v8.1.5 in Azure that I planned to upgrade to 8.15.3. As part of the upgrade, I took the opportunity to update some config as per the documentation on setting up load balanced environments in Azure - this meant adding a transform for Azure environments to insert the Umbraco.Core.MainDom.Lock setting with a value of SqlMainDomLock. I also wanted to take advantage of the new unattended install functionality, so I added the Umbraco.Core.RuntimeState.UpgradeUnattended setting to my master server.

On load of the master server, post deployment, I received a message that the website boot had failed. On viewing the logs, I was able to see the above errors and that the issue was with acquiring the MainDom. I found [this forum post](https://our.umbraco.com/forum/using-umbraco-and-getting-started/104664-upgrade-855-to-8101-cannot-acquire-maindom#comment-326641) and followed the advice to remove the Umbraco.Core.MainDom.Lock setting from web.config, which allowed me to proceed with the unattended upgrade. I then added the Umbraco.Core.MainDom.Lock setting back in to web.config and the website continued to boot as expected.

I had a look through issues and found the following, which refers specifically to obtaining MainDom in relation to unattended installs: https://github.com/umbraco/Umbraco-CMS/pull/10291. This feels related, but I'm not sure if it's v9 specific, because the failure details in the 'Test' section doesn't indicate the same error as I was seeing in v8.

My question is, is it expected that unattended upgrades won't work with this setting in place in v8, or have I gone wrong somewhere? I did test doing an unattended upgrade without the Umbraco.Core.MainDom.Lock setting in place and everything worked fine, so it feels as if there's an incompatibility between the two.

### Steps to reproduce

1. Install Umbraco 8.1.5 and deploy to Azure
2. Upgrade locally, via NuGet, to 8.15.3 and add the following web.config transforms: 
    1. Insert Umbraco.Core.MainDom.Lock with a value of SqlMainDomLock on both main and replica servers
    2. Insert Umbraco.Core.RuntimeState.UpgradeUnattended with a value of true on main server

3. Deploy the upgraded code to Azure
4. Access the main server in order to do the upgrade
5. Observe that booting has failed

### Expected result / actual result

Expected result: Unattended upgrade takes place
Actual result: Boot fails with MainDom errors in the logs

**Comment by vipervf1** [comment]
I observed this as well while upgrading from 8.3 to 8.15 on Azure.

I believe it's because pre-upgrade the main dom lock table does not yet exist in the database, so it fails to boot. 

What I ended up doing was deploying the upgrade without the setting and then applying it after. After doing this, applying a patch version of Umbraco with that setting enabled worked as expected.

**Comment by antonysmith-mando** [comment]
I ended up doing something similar, but I deployed once and updated the web.config on the server to remove the setting, ran the upgrade and then added it back in again. It was a client deployment so not so simple to tell them we needed to do two.

Good to hear that everything works as expected following the first time and, given your explanation of the issue, it makes sense why it would only be a one-time thing.

I think on that basis I'm happy to close this issue. It's not ideal, but sounds like a bit of an edge case scenario with a pretty easy workaround so probably not one that you'd expect a lot of time to be put into to resolve.
